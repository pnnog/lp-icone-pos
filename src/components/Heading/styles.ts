import styled, { css } from 'styled-components';
import { HeadingProps } from '.';
import media from 'styled-media-query';

export const Wrapper = styled.h2<HeadingProps>`
  ${({ theme, color }) => css`
    ${media.greaterThan('medium')`
    font-size: ${theme.font.sizes.xxlarge};
    `}

    display: flex;
    align-items: center;
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.bold};

    &::before {
      content: '';
      display: inline-block;
      margin-right: ${theme.spacings.xsmall};
      width: 0.7rem;
      height: 3.3rem;
      background-color: ${!!color && theme.colors[color]};
    }

    color: ${!!color && theme.colors[color]};
  `}
`;
