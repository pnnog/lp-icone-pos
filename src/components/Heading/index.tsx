import * as S from './styles';

export type HeadingProps = {
  children: React.ReactNode;
  color?: 'primary' | 'white' | 'orange';
  line?: 'left' | 'bottom';
};

const Heading = ({
  children,
  color = 'primary',
  line = 'left'
}: HeadingProps) => (
  <S.Wrapper line={line} color={color}>
    {children}
  </S.Wrapper>
);

export default Heading;
