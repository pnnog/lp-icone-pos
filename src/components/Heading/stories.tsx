import { Meta, Story } from '@storybook/react';
import Heading, { HeadingProps } from '.';

const props = {
  children: 'Corpo Docente'
};

export default {
  title: 'Heading',
  component: Heading,
  args: { ...props }
} as Meta;

export const Basic: Story<HeadingProps> = (args) => <Heading {...args} />;
