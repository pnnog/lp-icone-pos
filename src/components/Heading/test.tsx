import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helper';

import Heading from '.';

const props = {
  children: 'Corpo Docente'
};

describe('<Heading />', () => {
  // Deve renderizar o Heading com o texto desejado
  it('should render Heading', () => {
    renderWithTheme(<Heading {...props} />);
    expect(
      screen.getByRole('heading', { name: /Corpo Docente/i })
    ).toBeInTheDocument();
  });

  // Deve renderizar heading branco
  it('should render Heading with white color', () => {
    renderWithTheme(<Heading {...props} color="white" />);
    expect(screen.getByRole('heading', { name: /Corpo Docente/i })).toHaveStyle(
      { color: '#F8FAFC' }
    );
  });

  // Deve renderizar heading azul
  it('should render Heading with blue color', () => {
    renderWithTheme(<Heading {...props} />);

    expect(screen.getByRole('heading', { name: /Corpo Docente/i })).toHaveStyle(
      { color: '#0C4A6E' }
    );
  });

  //Deve renderizar heading small (media query)
  it('should render heading with small size', () => {
    renderWithTheme(<Heading {...props} />);
    expect(screen.getByRole('heading', { name: /Corpo Docente/i })).toHaveStyle(
      { 'font-size': '2.0rem' }
    );
  });

  // Deve renderizar heading normal size (media query)
  it('should render heading with normal size', () => {
    renderWithTheme(<Heading {...props} />);
    expect(
      screen.getByRole('heading', { name: /Corpo Docente/i })
    ).toHaveStyleRule('font-size', '2.8rem', {
      media: '(min-width:768px)'
    });
  });

  // Deve ter o cosmético branco
  it('should render heading with blue cosmetic ', () => {
    renderWithTheme(<Heading {...props} />);
    expect(
      screen.getByRole('heading', { name: /Corpo docente/i })
    ).toHaveStyleRule('background-color', '#0C4A6E', {
      modifier: '::before'
    });
  });

  // Deve ter o cosmético branco
  it('should render heading with white cosmetic ', () => {
    const { container } = renderWithTheme(<Heading {...props} color="white" />);
    expect(
      screen.getByRole('heading', { name: /Corpo docente/i })
    ).toHaveStyleRule('background-color', '#F8FAFC', {
      modifier: '::before'
    });

    expect(container).toMatchSnapshot();
  });
});
