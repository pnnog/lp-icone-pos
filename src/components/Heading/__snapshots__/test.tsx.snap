// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render heading with white cosmetic  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 2.0rem;
  font-weight: 600;
  color: #F8FAFC;
}

.c0::before {
  content: '';
  display: inline-block;
  margin-right: 1.6rem;
  width: 0.7rem;
  height: 3.3rem;
  background-color: #F8FAFC;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

<div>
  <h2
    class="c0"
    color="white"
  >
    Corpo Docente
  </h2>
</div>
`;
